src/ui/public/doc_table/components/table_row.js

```diff
@@ -92,18 +102,30 @@ module.directive('kbnTableRow', function ($compile) {
           openRowHtml
         ];
 
+        const mapping = indexPattern.fields.byName;
         if (indexPattern.timeFieldName) {
           newHtmls.push(cellTemplate({
             timefield: true,
-            formatted: _displayField(row, indexPattern.timeFieldName)
+            formatted: _displayField(row, indexPattern.timeFieldName),
+            filterable: mapping[indexPattern.timeFieldName].filterable,
+            column: indexPattern.timeFieldName
           }));
         }
 
         $scope.columns.forEach(function (column) {
+          const flattenedRow = indexPattern.flattenHit(row);
+          const isFilterable =
+            flattenedRow[column] === undefined
+            ? false
+            : !mapping[column]
+            ? false
+            : mapping[column].filterable;
           newHtmls.push(cellTemplate({
             timefield: false,
             sourcefield: (column === '_source'),
-            formatted: _displayField(row, column, true)
+            formatted: _displayField(row, column, true),
```

Comment:
Let's pass a `rawValue` param here (and for the timefield above) and pass it to the `inlineFilter` call in the template in the same way I suggested passing the column name. Then the call to `flattenHit` can move up to line 99, share the variable throughout `createSummaryRow` and it can be removed entirely from `inlineFilter` (since the field value will be passed as a param). This way flattenHit only needs to be called once per row.